packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -35,13 +41,64 @@ export const TranslationSidebar = ({
     ['BDB', 'LSJ'].includes(resource)
   );
   const lexiconEntry = lexiconResource?.entry ?? '';
-  const { t } = useTranslation(['common', 'translate']);
+
+  const translatorNotesQuery = useQuery(
+    ['verse-translator-notes', language, verse.id],
+    () => apiClient.verses.findTranslatorNotes(verse.id, language)
+  );
+  const translatorNote = translatorNotesQuery.isSuccess
+    ? translatorNotesQuery.data.data[word.id]
+    : null;
+  const originalNoteContent = translatorNote?.content ?? '';
 
   const tabTitles = ['translate:lexicon', 'translate:notes'];
   if (showComments) {
     tabTitles.push('translate:comments');
   }
 
+  const userCan = useAccessControl();
+
+  const canEditNote = userCan('translate', {
+    type: 'Language',
+    id: language,
+  });
+  const [noteAuthorName, setNoteAuthorName] = useState('');
+  const [noteTimestampString, setNoteTimestampString] = useState('');
+  const [noteContent, setNoteContent] = useState('');
+
+  useEffect(() => {
+    setNoteAuthorName(translatorNote?.authorName ?? '');
+    setNoteTimestampString(
+      translatorNote?.timestamp
+        ? new Date(translatorNote?.timestamp).toLocaleString()
+        : ''
+    );
+    setNoteContent(originalNoteContent);
+  }, [translatorNote, originalNoteContent]);
```

Comment:
Let's try modifying this effect in a few ways:
1. author name and timestamp don't need to be a separate state. They can just be computed like normal: `const noteAuthorName = translatorNote?.authorName ?? ''`. When you can, derive state, don't sync it.
2. However, for the note content, I think we do need to sync state. We can keep a ref of the current word. And only update the note content if that ref indicates that the word has changed. That means we need to replace the rich text editor with that word's note.

I think with these two changes, the react state will always reflect the latest thing that the user has typed, but the author and timestamp will be derived from the latest server state.

Reply:
Ok, I think I implemented this like you describe. It seems to work now! I had to add the `key` in https://github.com/arrocke/gloss-translation/pull/321/commits/52e3c8bdf185897d5dfb2e53fbe65bbec35cdd7a to resolve a bug where clicking on a word that didn't have any note didn't correctly clear out the rich text input.